/**
 * Copyright (c) 2022 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";
package protoPanda;

message VariantValue {
    enum VariantValueType {
        BOOL = 0;
        U8 = 1;
        U16 = 2;
        U32 = 3;
        U64 = 4;
        F32 = 5;
        F64 = 6;
        STRING = 7;
    }

    oneof value {
        uint64 valueInt = 1;
        float valueFloat = 2;
        double valueDouble = 3;
        bytes valueStr = 4;
    }
    VariantValueType type = 5;
}

message Literal {
    uint32 tag = 1; // LiteralTag
    VariantValue value = 2;
}

message LiteralArray {
    repeated Literal literals = 1;
}
